dozer/cogs/_utils.py

```diff
@@ -289,4 +299,56 @@ def decorator(func):
             func.__required_permissions__ = discord.Permissions()
             func.__required_permissions__.update(**required)
         return func
+
     return decorator
+
+
+class PrefixHandler:
+    """Handles dynamic prefixes"""
+
+    def __init__(self, default_prefix):
+        self.default_prefix = default_prefix
+        self.prefix_cache = {}
+
+    def handler(self, bot, message):
+        """Process the dynamic prefix for each message"""
+        dynamic = self.prefix_cache.get(message.guild.id if message.guild else 0)
```

Comment:
I feel like this should check if the message is not in a guild first and return it's own error rather than making the assumption that a guild_id of 0 will always return a `None`. While I can't see any way that gets written at this point, explicitly checking it would be much cleaner and likely more performant too.